src/drivers/driver_zephyr.c

```diff
@@ -1507,7 +1507,7 @@ static int wpa_drv_zep_set_supp_port(void *priv,
 				     authorized,
 				     if_ctx->bssid);
 
-#ifdef CONFIG_NET_DHCPV4
+#if !defined(CONFIG_AP) && defined(CONFIG_NET_DHCPV4)
```

Comment:
This should be `CONFIG_STA` mode (need to introduce it), as we might have other cases like P2P without AP.

Reply:
I see that CONFIG_P2P selects CONFIG_AP. So, this check should also cover P2P case as well, right?

Reply:
Removed the config dependencies that were added here.